### ISSUE TITLE
operationId with dashes

### ISSUE THREAD

**Issue by j12y** [readmeio/api#507]
We have an Open API Spec with a setting such as:

```json
"operationId":"get-api-token",
```

This generates a function such as:

```javascript
const sdk = require('api')('...');

sdk.auth('...', '...');
sdk.getApiToken({grant_type: 'client_credentials'}, {
  Accept: 'application/json',
  'Cache-Control': 'no-cache'
})
  .then(res => console.log(res))
  .catch(err => console.error(err));
```

This works fine in the Readme.com reference documentation.  If you install the `api` library locally however it generates an error.

> Error: Sorry, `getApiToken` does not appear to be a valid operation on this API.

Editing the code to instead use `sdk['get-api-token']` works as expected:

```
< sdk.getApiToken({grant_type: 'client_credentials'}, {
---
> sdk['get-api-token']({grant_type: 'client_credentials'}, {
```

**Comment by erunion** [comment]
Which version of api are you seeing this on?

**Comment by j12y** [comment]
```
$ grep version node_modules/api/package.json
  "version": "4.5.1",
```

**Comment by erunion** [comment]
Ok I see the problem, I fixed a bunch of issues within `api` to clean up how we handle operation IDs but I shipped those to our current beta of v5 -- and then updated our snippet generation in production to use these new changes to operation IDs.

I'll backport these fixes today and push out a v4.5.2 release. Thanks and apologies for discovering this.

**Comment by j12y** [comment]
You've been very responsive and helpful, thanks for looking into it and getting to the bottom of it.

**Comment by erunion** [comment]
@j12y I've just published a [v4.5.2 release](https://github.com/readmeio/api/releases/tag/v4.5.2) with some fixes to this. Though we'll continue to show `getApiToken` as being the SDK accessor in code snippets (as it functions more as a method accessor) on your ReadMe project, `getApiToken` and `get-api-token` will now both work as expected.
